src/vdm.c

```diff
@@ -12,14 +12,17 @@ struct vdm {
 };
 
 /*
- * vdm_new -- returns NULL if failed to allocate memory for struct vdm
- * or vdm_data_init failed
+ * vdm_new -- returns NULL if descriptor is NULL, failed to allocate memory
+ * for struct vdm or vdm_data_init failed.
  */
 struct vdm *
 vdm_new(struct vdm_descriptor *descriptor)
 {
+	if (descriptor == NULL)
```

Comment:
No, this type of check is almost never a good idea. Errors should be caught at the earliest possible moment. This NULL-check should be resolved at the call site.